### PR TITLE
sampling: added a test verifying manual keep is having precedence over rules

### PR DESCRIPTION
**Pull request by dianashevchenko**
## Motivation

We had a report of spans being dropped despite being marked as 'manual.keep'. In that case, the tracing rules were overriding, which shouldn't be the case. It was fixed in Go and Java.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Comment by cbeauchesne** [comment]
Hi @dianashevchenko  , do you still plan to work on this?

**Comment by dianashevchenko** [comment]
@cbeauchesne hi yes, sorry. It fell through the cracks

File: utils/build/docker/golang/parametric/go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	go.opentelemetry.io/otel/trace v1.20.0
 	google.golang.org/grpc v1.57.1
 	google.golang.org/protobuf v1.30.0
-	gopkg.in/DataDog/dd-trace-go.v1 v1.39.0-alpha.1.0.20240419091955-7df262e804ad
+	gopkg.in/DataDog/dd-trace-go.v1 v1.39.0-alpha.1.0.20240424121809-d9b1b369853f
```
Comment (darccio) [inline utils/build/docker/golang/parametric/go.mod:10]: Why is this using this oddly old `dd-trace-go` version?

**Review by darccio — CHANGES_REQUESTED** [review]

File: utils/build/docker/golang/parametric/go.mod (hunk 1 above)
Comment (dianashevchenko) [reply utils/build/docker/golang/parametric/go.mod:10]: It's actually not, but that's how the version is displayed

**Review by darccio — APPROVED** [review]

**Review by mtoffl01 — APPROVED** [review]

File: tests/parametric/test_trace_sampling.py [hunk 2]
```diff
@@ -77,6 +77,29 @@ def test_trace_dropped_by_trace_sampling_rule(self, test_agent, test_library):
...
+                    [{"service": "webserver", "resource": "drop-me", "sample_rate": 0}]
+                ),
+            }
+        ],
+    )
+    def test_trace_kept_in_spite_trace_sampling_rule(self, test_agent, test_library):
+        """Test that a trace is being kept with manual.keep depite of the matching defined trace sampling rule"""
+        with test_library:
+            with test_library.start_span(name="web.request", service="webserver") as s1:
+                s1.set_metric("sampling.priority", 2)
```
Comment (lucaspimentel) [inline tests/parametric/test_trace_sampling.py:96]: Setting this tag directly is not a standard way of setting "manual keep" and won't necessarily work for all languages. For example, in .NET we have a `SetSamplingPriority()` API which does more than just setting this tag.

Comment (lucaspimentel) [inline tests/parametric/test_trace_sampling.py:96]: This way of setting sampling priority by users was deprecated a long time ago in favor of setting `"manual.keep"`.

**Pull request merged**